app/src/main/java/com/eveningoutpost/dexdrip/Services/LibreWifiReader.java

```diff
@@ -57,6 +58,9 @@
     private static final Gson gson = JoH.defaultGsonInstance();
 
     private final static long DEXCOM_PERIOD = 300000;
+    
+    // An object that is used to sync the readData
+    private static final Object readDataLock = new Object();
```

Comment:
Can you just do: `synchronized (LibreWifiReader.class)` to avoid creating an additional object?

Reply:
Done.